dataeng/resources/tableau-config-backup.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -ex
+
+ssh -tt $USER_NAME@$TABLEAU_SERVER_IP -o StrictHostKeyChecking=no "S3_BUCKET=$S3_BUCKET" '
+
+# It gets the current date from system in 2022.04.05-10:35 format
+DATE="$(date '+%Y.%m.%d')"
+
+# It adds the date as prefix on backup name. Backup runs at 09:00 UTC so it also prefixed in the name.
+BACKUPNAME="${DATE}-tableau-config-backup.json"
+
+sudo runuser -l tsm_admin -c "tsm settings export -f /home/tableau_backups/tableau-config-backup.json"
+
+# Check if tableau backup file exist, exit if the file does not exist
+if [[ ! -f /home/tableau_backups/tableau-config-backup.json ]]
+then
+    echo "$BACKUPNAME Backup file does not exist. It seems to be some problem generating tableau backup. Refer to Administration of Tableau wiki docs to debug Tableau Regular Backup Automation"
+    exit 1
+else
+    echo "Copying backup file into s3"
+    aws s3 cp /home/tableau_backups/tableau-config-backup.json s3://$S3_BUCKET/$BACKUPNAME
+
+    echo "Removing backup file from local disk"
+    sudo rm -f /home/tableau_backups/$BACKUPNAME
+
+fi
+
+'
+
+# Creating python 3.8 virtual environment to run python script.
+PYTHON38_VENV="py38_venv"
+virtualenv --python=python3.8 --clear "${PYTHON38_VENV}"
+source "${PYTHON38_VENV}/bin/activate"
+
+DATE="$(date '+%Y.%m.%d')"
+TABLEAU_CONFIG=${DATE}-tableau-config-backup.json
```

Comment:
This is defined twice. Maybe just define it once and pass it into the ssh script as a variable?